### PR TITLE
B4a 424 - Adds temp unique User control.

### PR DESCRIPTION
**Pull request by seijiakiyama**
(This will be implemented as "unique" attribute to all Entities in a near future)

### CODE REVIEW

File: src/back/entities/entityRouter.js [hunk 1]
```diff
@@ -115,37 +115,45 @@ function postEntity(entities) {
...
-      .catch(function () {
-        res.status(500).json({
-          code: 1,
-          error: 'Internal Server Error'
         });
+    })
+    .catch(function () {
+      res.status(400).json({
+        code: 104,
+        error: 'Duplicated Entry'
```
Comment (brunopicinin) [inline src/back/entities/entityRouter.js]: The correct error message is "Duplicate Entity"

File: src/back/entities/entityRouter.js [hunk 2]
```diff
@@ -357,35 +365,42 @@ function updateEntity(entities) {
...
+                        error: 'Internal Server Error'
+                      });
+                    }
+                  });
+              });
             });
+        }).catch(function () {
+          res.status(400).json({
+            code: 104,
+            error: 'Duplicated Entry'
```
Comment (brunopicinin) [inline src/back/entities/entityRouter.js]: Duplicate Entity

Comment (brunopicinin) [inline src/back/entities/entityRouter.js]: The correct error message is "Duplicate Entity"

File: src/back/entities/entityRouter.js [hunk 3]
```diff
@@ -573,6 +588,32 @@ function _replacePermissionsInUser(entity, entityName) {
...
 
+function _checkDuplicatedUsername(entity, entityName) {
+  return new Promise(function (resolve, reject) {
+    // only change users
+    if (entityName !== 'User') {
+      resolve();
+      return;
+    }
+
+    var orArray = [
```
Comment (brunopicinin) [inline src/back/entities/entityRouter.js]: No need to break link, nor use quotation marks on the username key.

File: src/back/entities/entityRouter.js [hunk 4]
```diff
@@ -573,6 +588,32 @@ function _replacePermissionsInUser(entity, entityName) {
...
+      {'username': entity.username}];
+    if (entity.email) {
+      orArray.push({'email': entity.email});
+    }
+
+    // searchs for other occurrences
```
Comment (brunopicinin) [inline src/back/entities/entityRouter.js]: "searches"

Comment (seijiakiyama) [inline src/back/entities/entityRouter.js]: Maaaan, those eyes are sharp as Tramontina's knives...

**Pull request merged**